x-pack/plugins/endpoint/server/routes/alerts/details/handlers.ts

```diff
@@ -22,22 +22,32 @@ export const alertDetailsHandlerWrapper = function(
     try {
       const alertId = req.params.id;
       const response = (await ctx.core.elasticsearch.dataClient.callAsCurrentUser('get', {
+        // Remove the hard coded reference to an alert index here
+        // https://github.com/elastic/endpoint-app-team/issues/311
         index: EndpointAppConstants.ALERT_INDEX_NAME,
         id: alertId,
       })) as GetResponse<AlertEvent>;
 
+      const indexPattern = await endpointAppContext.indexPatternRetriever.get(
+        ctx.core.savedObjects.client,
+        EndpointAppConstants.EVENT_DATASET
+      );
+
```

Comment:
Is this constant used anywhere apart from pattern retrieval. Will it be better to have `indexPatternRetriever.getEventIndexPattern(ctx.core.savedObjects.client)`. And hide the implementation details.

Reply:
Yeah good point. I'll make the changes you suggested.

Reply:
Looking at this closer now. `EVENT_DATASET` is also used by the frontend code: https://github.com/jonathan-buttner/kibana/blob/endpoint-ingest-index-pattern/x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts#L19

So the frontend needs some constant to indicate which index pattern it is trying to retrieve. I'll make the changes you suggested though for the places where server side code retrieves the index pattern.